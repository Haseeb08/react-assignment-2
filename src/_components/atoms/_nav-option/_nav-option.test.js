import React from "react";
import { create } from "react-test-renderer";
import { render, screen, getByRole,container, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import NavItem from "./_nav-option";

describe("Nav item testing", () => {
  test("Matches the snapshot", () => {
    const item = create(<NavItem />);
    expect(item.toJSON()).toMatchSnapshot();
  });

  test("Nav item  is present", () => {
    // Using screen
    const {getByTestId,container } = render(<NavItem title={"home"} />);


   //const item = screen.getByTestId("input");

    expect(getByTestId("input")).toBeInTheDocument();
    fireEvent.mouseOver(getByTestId("input"));
    expect(container.firstChild).toHaveTextContent("home");
  });
});
