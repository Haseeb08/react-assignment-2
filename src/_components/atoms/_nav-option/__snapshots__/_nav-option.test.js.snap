// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav item testing Matches the snapshot 1`] = `
<div
  className="makeStyles-itemBox-1"
  data-testid="input"
>
  <h4
    className="MuiTypography-root MuiTypography-h4"
  >
    Name
  </h4>
</div>
`;
